src/composables/useLoop/index.ts

```diff
@@ -0,0 +1,67 @@
+import type { EventDispatcher, Raycaster, WebGLRenderer } from 'three'
+import { unref } from 'vue'
+import type { TresCamera, TresScene } from '../../types'
+import { useTresContext } from '../useTresContextProvider'
+import type { LoopCallback } from '../../core/loop'
+import type { SizesType } from '../useSizes'
+
+export interface LoopParams extends LoopCallback {
+  camera: TresCamera
+  scene: TresScene
+  renderer: WebGLRenderer
+  invalidate: () => void
+  advance: () => void
+  raycaster: Raycaster
+  controls: (EventDispatcher & { enabled: boolean })
+  sizes: SizesType
+}
+
+export type Fn = (params: LoopParams) => void
+
+export function useLoop() {
+  const {
+    camera,
+    scene,
+    renderer,
+    loop,
+    raycaster,
+    controls,
+    invalidate,
+    advance,
+  } = useTresContext()
+
+  function wrapCallback(cb: Fn) {
+    return (params: LoopCallback) => {
+      cb({ ...params, camera: unref(camera) as TresCamera, scene: unref(scene), renderer: unref(renderer), raycaster: unref(raycaster), controls: unref(controls), invalidate, advance })
+    }
+  }
+
+  function onBeforeRender(cb: Fn, index = 0) {
+    const wrappedCallback = wrapCallback(cb)
+    const { off } = loop.register(wrappedCallback, 'before', index)
+    return { off }
+  }
+
+  function render(cb: Fn) {
+    const wrappedCallback = wrapCallback(cb)
+    const { off } = loop.register(wrappedCallback, 'render')
+    return { off }
```

Comment:
This can't be `off`ed.

```playground/src/components/TakeOverLoopExperience.vue```

```html
<script setup lang="ts">
import { useLoop } from '@tresjs/core'

import { OrbitControls } from '@tresjs/cientos'
import { useControls } from '@tresjs/leches'

const { render, pauseRender, resumeRender } = useLoop()

const off = render(({ renderer, scene, camera }) => {
  console.log('on')
  renderer.render(scene, camera)
}).off

setTimeout(() => {off(); console.log('offing')}, 1000)

const { isRenderPaused } = useControls({
  isRenderPaused: {
    value: false,
    type: 'boolean',
    label: 'Pause Render',
  },
})

watchEffect(() => {
  if (isRenderPaused.value) {
    pauseRender()
  }
  else {
    resumeRender()
  }
})
</script>

<template>
  <TresPerspectiveCamera :position="[3, 3, 3]" />
  <OrbitControls />
  <AnimatedObjectUseUpdate />
  <TresAmbientLight :intensity="1" /> />
  <TresGridHelper />
</template>
```

Reply:
Hey @andretchen0  Check the discord discussion